tests/test_site_apportionment/test_site_apportionment.py

```diff
@@ -634,3 +636,127 @@ def test_sort_rows_missing_cols(self, caplog):
 
         with pytest.raises(KeyError):
             sort_rows_order_cols(input_df, cols_in_order)
+
+
+class TestWeightValues(object):
+    """Tests for weight_values."""
+    
+    @pytest.fixture(scope="function")
+    def weight_values_test_df(self):
+        """Test data for weight_values."""
+        frame = {
+            "reference": [0, 1, 2, 3, 4],
+            "val_col_1": [10, 14, 16, 18, 6],# test using ints rather than float
+            "val_col_2": [3.5, 4.5, 10, np.nan, 0],# test np.nan+0+floats
+            "weight_col": [1.5, 1.5, 1.0, 2.0, 3.0]
+                }
+        return pandasDF(frame)
+    
+    @pytest.fixture(autouse=True)
+    def set_attrs(self):
+        """Set class attributes to pass to function."""
+        self.value_cols = ["val_col_1", "val_col_2"]
+        self.weight_col = "weight_col"
+
+    def test_weight_values_raises(self, weight_values_test_df):
+        """Tests for raises in weight_values."""
+        with pytest.raises(KeyError, match=".*not_weight_col.*"):
+            weight_values(weight_values_test_df, 
+                          self.value_cols, 
+                          "not_weight_col")
```

Comment:
Can you explain why this is a good idea? Could we leave it out?

Reply:
This ensure that the function does not work without a required column.